### PR TITLE
Pr stage2unpacker 80x

### PR DESCRIPTION
**Pull request by mulhearn**
This PR adds all of the L1T raw data packers and unpackers that use the L1TRawToDigi framework.  It Includes the most crucial unpackers needed for the 2016 run (Calo Stage2, uGMT, and uGT).  It includes also the EMTF and BMTF unpacker, which are not yet included in the standard sequence.

This PR will become smaller as #13065 is merged.

### CODE REVIEW

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
I can rebase to remove ALCA/DB now...

**Comment by mulhearn** [comment]
(Rebased, no more ALCA/DB sign off needed)

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
@davidlange6  FYI this one is the next in the L1 pipeline.  Hahhaha... thats a DAQ joke.  Get it?

File: DataFormats/L1DTTrackFinder/interface/L1MuDTChambPhDigi.h [hunk 1]
```diff
@@ -41,6 +41,9 @@ class L1MuDTChambPhDigi {
   L1MuDTChambPhDigi( int ubx, int uwh, int usc, int ust,
 	int uphr, int uphb, int uqua, int utag, int ucnt );
 
+  L1MuDTChambPhDigi( int ubx, int uwh, int usc, int ust,
+        int uphr, int uphb, int uqua, int utag, int ucnt, int urpc);
+
```
Comment (davidlange6) [inline DataFormats/L1DTTrackFinder/interface/L1MuDTChambPhDigi.h]: @mulhearn - remove the old constructor and add a default of -10 for int urpc

File: EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
-    HFRingSumsInputLabel = cms.InputTag("caloStage1Digis", "HFRingSums"),
-    RegionInputLabel = cms.InputTag("caloStage1Digis", ""),
-    EmCandInputLabel = cms.InputTag("caloStage1Digis", ""),
+    InputLabel = cms.InputTag("simCaloStage1FinalDigis"),
+    TauInputLabel = cms.InputTag("simCaloStage1FinalDigis", "rlxTaus"),
+    IsoTauInputLabel = cms.InputTag("simCaloStage1FinalDigis", "isoTaus"),
+    HFBitCountsInputLabel = cms.InputTag("simCaloStage1FinalDigis", "HFBitCounts"),
+    HFRingSumsInputLabel = cms.InputTag("simCaloStage1FinalDigis", "HFRingSums"),
+    RegionInputLabel = cms.InputTag("simRctDigis", ""),
+    EmCandInputLabel = cms.InputTag("simRctDigis", ""),
```
Comment (davidlange6) [inline EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py]: @mulhearn - why this change?

File: EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
     Setup           = cms.string("stage2::GTSetup"),
     InputLabel      = cms.InputTag("rawDataCollector"),
     FedIds          = cms.vint32( 1404 ),
-    FWId            = cms.untracked.int32(2),
+    FWId            = cms.uint32(2),
     lenSlinkHeader  = cms.untracked.int32(8),
     lenSlinkTrailer = cms.untracked.int32(8),
```
Comment (davidlange6) [inline EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py]: likely all of these parameters should be tracked..

File: EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py (hunk 2 above)
Comment (mulhearn) [inline EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py]: Actually, that looks wrong... why didn't this break anything?

File: EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py (hunk 3 above)
Comment (mulhearn) [inline EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py]: Indeed... but my attempt to changing some of the other ones broke HLT scripts.  So was postponing this until HLT/L1T integration step.

File: EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py (hunk 2 above)
Comment (mulhearn) [inline EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py]: Our open development branch model leads to some surprises like this slipping by...

File: DataFormats/L1DTTrackFinder/interface/L1MuDTChambPhDigi.h (hunk 1 above)
Comment (mulhearn) [inline DataFormats/L1DTTrackFinder/interface/L1MuDTChambPhDigi.h]: OK, will update.

**Comment by mulhearn** [comment]
Let's hold off on this one until I can make your suggested changes...  give me a few.

File: EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py (hunk 2 above)
Comment (mulhearn) [inline EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py]: Sorry, no actually, that is a correct correction.  We produce RAW from simulated digis (or the detector does it for us).  So the old version was a mistake.  This one is correct.

**Comment by mulhearn** [comment]
OK, there's the update.

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
please test
